### PR TITLE
stdio/format: Add support for hh, h and t length modifiers

### PR DESCRIPTION
**Pull request by astalke**
- Add dummy support for 'L' length modifier (long double is cast to double)
- Add support for 'h' and 'hh' length modifiers
- Add support for 't' length modifier

DONE: RTOS-356

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: stdio/format.c [hunk 1]
```diff
@@ -509,6 +549,7 @@ static void format_sprintf_num(void *ctx, feedfunc feed, uint64_t num64, uint32_
 
 void format_parse(void *ctx, feedfunc feed, const char *format, va_list args)
 {
+	double double_number;
```
Comment (agkaminski) [inline stdio/format.c:552]: Use camelCase (I am aware of existing violation of this style in this file)

File: stdio/format.c [hunk 2]
```diff
@@ -35,21 +35,53 @@
...
 #define FLAG_FIELD_WIDTH_STAR 	0x2000
 #define FLAG_DOUBLE		   0x4000
+#define FLAG_8BIT               0x8000
+#define FLAG_16BIT              0x10000
+#define FLAG_LONG_DOUBLE        0x20000
+
+
+#define GET_UNSIGNED(number, flags, args) \
+	do { \
+		if (((flags)&FLAG_8BIT) != 0) { \
```
Comment (agkaminski) [inline stdio/format.c:45]: clang format is confused, applies in other places too
```suggestion
		if (((flags) & FLAG_8BIT) != 0) { \
```

Comment (agkaminski) [reply stdio/format.c:45]: You can use `/* clang-format off */` and `/* clang-format on */`

**Review by agkaminski — APPROVED** [review]

**Comment by niewim19** [comment]
Referencing connected issues:
 - https://github.com/phoenix-rtos/phoenix-rtos-project/issues/277

**Review by agkaminski — APPROVED** [review]

**Review by damianloew — COMMENTED** [review]
Please rebase this branch, above that I have tested those changes on armv7m4-stm32l4x6-nucleo and it looks good, so I will approve it after rebasing.

**Review by damianloew — APPROVED** [review]

**Pull request merged**